### PR TITLE
Adding a CLI Test: Execute a 1.0 tool in a netcoreapp1.1 project

### PR DESCRIPTION
**Pull request by johnbeisner**
https://github.com/dotnet/cli/issues/6229
@dotnet/dotnet-cli

Last of a planned set of backward compat tests for the CLI given that we are changing the CLI/SDK 'look-up' policy: https://github.com/dotnet/cli/issues/6038

Related to:
https://github.com/dotnet/cli/issues/6228
https://github.com/dotnet/cli/issues/6227

### CODE REVIEW

File: TestAssets/TestProjects/TestAppWithCLIToolReferences/TestAppWithCLIToolReferences.csproj [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), testAsset.props))\testAsset.props" />
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+    <RuntimeFrameworkVersion>$(CLI_SharedFrameworkVersion)</RuntimeFrameworkVersion>
```
Comment (livarcocc) [inline TestAssets/TestProjects/TestAppWithCLIToolReferences/TestAppWithCLIToolReferences.csproj]: This should not be here.

File: TestAssets/TestProjects/TestAppWithCLIToolReferences/TestAppWithCLIToolReferences.csproj [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    <DotnetCliToolTargetFramework>netcoreapp1.1</DotnetCliToolTargetFramework>
```
Comment (livarcocc) [inline TestAssets/TestProjects/TestAppWithCLIToolReferences/TestAppWithCLIToolReferences.csproj]: This shouldn't be here either. Current 1.0 projects don't have this set.

**Comment by livarcocc** [comment]
Use this attribute in your test: `[RequiresSpecificFrameworkFact("netcoreapp1.0")]` to prevent it from running on distros that don't support netcoreapp1.0.

**Comment by livarcocc** [comment]
@ManishJayaswal This is a test only change. Can you approve.

File: TestAssets/TestProjects/TestAppWithCLIToolReferences/TestAppWithCLIToolReferences.csproj (hunk 1 above)
Comment (johnbeisner) [inline TestAssets/TestProjects/TestAppWithCLIToolReferences/TestAppWithCLIToolReferences.csproj]: done...

File: TestAssets/TestProjects/TestAppWithCLIToolReferences/TestAppWithCLIToolReferences.csproj (hunk 2 above)
Comment (johnbeisner) [inline TestAssets/TestProjects/TestAppWithCLIToolReferences/TestAppWithCLIToolReferences.csproj]: done...

**Comment by ManishJayaswal** [comment]
I believe test only changes don't need approval.

File: TestAssets/TestProjects/TestAppWithCLIToolReferences/TestAppWithCLIToolReferences.csproj [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (livarcocc) [inline TestAssets/TestProjects/TestAppWithCLIToolReferences/TestAppWithCLIToolReferences.csproj]: You should probably call this test asset 10TestAppWith10CLIToolReferences

Comment (livarcocc) [inline TestAssets/TestProjects/TestAppWithCLIToolReferences/TestAppWithCLIToolReferences.csproj]: Once that's fixed. This LGTM.

**Pull request merged**